### PR TITLE
PMM-9055 Configurable query length for Postgres.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
PMM-9055

Build: [SUBMODULES-9055](https://github.com/Percona-Lab/pmm-submodules/pull/2787)

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

**Review by percona-csalguero — APPROVED** [review]

File: admin/commands/inventory/add_agent_qan_postgres_pgstatements_agent.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ type AddAgentQANPostgreSQLPgStatementsAgentCommand struct {
 	Password            string            `help:"PostgreSQL password for QAN agent"`
 	CustomLabels        map[string]string `mapsep:"," help:"Custom user-assigned labels"`
 	SkipConnectionCheck bool              `help:"Skip connection check"`
+	QueryLength         int32             `placeholder:"NUMBER" help:"Limit query length in QAN (default: server-defined; -1: no limit)"`
```
Comment (artemgavrilov) [inline admin/commands/inventory/add_agent_qan_postgres_pgstatements_agent.go:55]: I suggest to name this field `MaxQueryLength` to make it clear that it defines limitation.

File: agent/agents/postgres/pgstatmonitor/pgstatmonitor.go [hunk 2]
```diff
@@ -49,12 +49,14 @@ type PGStatMonitorQAN struct {
 	l                    *logrus.Entry
 	changes              chan agents.Change
 	monitorCache         *statMonitorCache
+	queryLength          int
```
Comment (artemgavrilov) [inline agent/agents/postgres/pgstatmonitor/pgstatmonitor.go:52]: Why not `int32`?

**Review by artemgavrilov — APPROVED** [review]

File: admin/commands/inventory/add_agent_qan_postgres_pgstatements_agent.go (hunk 1 above)
Comment (JiriCtvrtka) [reply admin/commands/inventory/add_agent_qan_postgres_pgstatements_agent.go]: Done for this PR and PR for MySQL.

File: agent/agents/postgres/pgstatmonitor/pgstatmonitor.go (hunk 2 above)
Comment (JiriCtvrtka) [reply agent/agents/postgres/pgstatmonitor/pgstatmonitor.go]: https://github.com/percona/pmm/pull/1216#discussion_r978350785

**Pull request merged**